pkg/controllers/nodeclaim/termination/controller.go

```diff
@@ -89,14 +91,41 @@ func (c *Controller) Finalize(ctx context.Context, nodeClaim *v1beta1.NodeClaim)
 		return reconcile.Result{}, nil
 	}
 	if nodeClaim.Status.ProviderID != "" {
-		if err = c.cloudProvider.Delete(ctx, nodeClaim); cloudprovider.IgnoreNodeClaimNotFoundError(err) != nil {
-			// We expect cloudProvider to emit a Retryable Error when the underlying instance is not terminated and if that
-			// happens, we want to re-enqueue reconciliation until we terminate the underlying instance before removing
-			// finalizer from the nodeClaim.
-			if cloudprovider.IsRetryableError(err) {
-				return reconcile.Result{RequeueAfter: 10 * time.Second}, nil
+		// Check if the status condition on nodeClaim is Terminating
+		if nodeclaimutil.GetCondition(nodeClaim, v1beta1.Terminating).Status != v1.ConditionTrue {
+			// If not then call Delete on cloudProvider to trigger termination
+			if err = c.cloudProvider.Delete(ctx, nodeClaim); cloudprovider.IgnoreNodeClaimNotFoundError(err) != nil {
+				return reconcile.Result{}, fmt.Errorf("terminating cloudprovider instance, %w", err)
+			}
+			// Set the status condition on nodeClaim as Terminating
+			nodeClaim.StatusConditions().SetCondition(apis.Condition{
+				Type:     v1beta1.Terminating,
+				Status:   v1.ConditionTrue,
+				Severity: apis.ConditionSeverityInfo,
+			})
+			if !equality.Semantic.DeepEqual(stored, nodeClaim) {
+				// We call Update() here rather than Patch() because patching a list with a JSON merge patch
+				// can cause races due to the fact that it fully replaces the list on a change
+				// https://github.com/kubernetes/kubernetes/issues/111643#issuecomment-2016489732
+				if err = c.kubeClient.Status().Update(ctx, nodeClaim); err != nil {
+					if errors.IsConflict(err) {
+						return reconcile.Result{Requeue: true}, nil
+					}
+					return reconcile.Result{}, client.IgnoreNotFound(err)
+				}
 			}
-			return reconcile.Result{}, fmt.Errorf("terminating cloudprovider instance, %w", err)
+			// Requeue nodeClaim termination
+			return reconcile.Result{RequeueAfter: 10 * time.Second}, nil
```

Comment:
Since we're doing `Get()` operations, can we drop this requeue interval down now?